apps/plugin-browser/src/shared/storage.ts

```diff
@@ -124,6 +155,64 @@ export const setInLocalStorage = async (
       setDisabledBadge();
     }
   }
+
+  /**
+   * Persist local storage state to the database where we want to preserve state across devices/browsers/log-outs
+   */
+  if (!skipDbPersist && isDbPersistedSetting(key)) {
+    const user = await getFromLocalStorage("user");
+    const settingsEntityId = user?.settingsEntityId;
+    if (!settingsEntityId) {
+      throw new Error("User somehow has no browser plugin settings entity");
+    }
+
+    const currentAutomaticConfig = await getFromLocalStorage(
+      "automaticInferenceConfig",
+    );
+    const currentManualConfig = await getFromLocalStorage(
+      "manualInferenceConfig",
+    );
+    const currentPopupTab = await getFromLocalStorage("popupTab");
+    const currentDraftNote = await getFromLocalStorage("draftQuickNote");
```

Comment:
I'm wondering whether this could be problematic because the async calls are happening outside of the debounced function. If I'm understanding right, it would be possible for a user to make changes in quick succession, and for an older change to result in the most recent call to the debounce function, which would make it possible for stale data to be persisted in the db. Granted this is probably quite unlikely, but if it's not a time consuming change maybe it's worth adding these async calls and the creation of the `newProperties` object to the body of `debouncedEntityUpdate`?

Reply:
I think whether or not this is technically possible depends on how `browser.storage.get` works, and whether it's possible for the calls to return in a different order than they entered the call stack. I've moved the calls inside the function inhttps://github.com/hashintel/hash/pull/3864/commits/53d5a29660ce76311b08bbd70b33cb7faf8ad929 in any case as it doesn't hurt and will avoid the problem you describe, if it's possible